transcendence-app/src/user/user.service.ts

```diff
@@ -250,4 +249,80 @@ export class UserService {
     }
     return friends;
   }
+
+  async getUserWithRolesFromUsername(
+    username: string,
+  ): Promise<UserWithRoles | null> {
+    const user = await this.userRepository.getByUsername(username);
+    if (!user) {
+      throw new UnprocessableEntityException();
+    }
+    const userWithRoles = await this.userToRoleRepository.getUserWithRoles(
+      user.id,
+    );
+    if (!userWithRoles) {
+      throw new UnprocessableEntityException();
+    }
+    return userWithRoles;
+  }
+
+  async getUserWithRolesFromId(id: string): Promise<UserWithRoles | null> {
+    const userWithRoles = await this.userToRoleRepository.getUserWithRoles(id);
+    if (!userWithRoles) {
+      throw new UnprocessableEntityException();
+    }
+    return userWithRoles;
+  }
+
+  async addUserToRole(user: UserToRole): Promise<UserToRole | null> {
+    const userToRole = await this.userToRoleRepository.addUserToRole(user);
+    if (!userToRole) {
+      throw new UnprocessableEntityException();
+    }
+    return userToRole;
+  }
+
+  async addUserToRoleFromUsername(
+    username: string,
+    role: string,
```

Comment:
```suggestion
    role: Role,
```

suggestion: If I'm not mistaken, we could use the `Role` type here and in the `deleteUserToRoleFromUsername()` method

Reply:
I'm not sure about this (I'd prefer the calling method to not have to be aware of the Role type), but I don't mind much anyway, so it's done. I'll keep changing everything in the next PR.